### PR TITLE
Client CLI module docs

### PR DESCRIPTION
**Pull request by emostov**
To view this in rustdoc form run

```
cargo doc --open -p qos-client
```

This should take a second to build and then pop open a new tab in your default web browser. Once its opens click on `cli` under the _Modules_ header.

### CODE REVIEW

File: qos-client/src/cli/mod.rs [hunk 1]
```diff
@@ -1,6 +1,263 @@
...
+//!
+//! ## Quorum Key Generation
+//!
+//! ### Generate Setup Keys
+//!
+//! The enclave has a special "genesis" service to generate a quorum key and
+//! shard it across the quorum members.
+//!
+//! For each member of the Quorum Set, the genesis service needs a corresponding
+//! Setup Key as input. (This key is used to encrypt the member specific outputs
```
Comment (jack-kearney) [inline qos-client/src/cli/mod.rs]: I think the justification for setup key is kind of confusing so would be good to add a little more detail here

File: qos-client/src/cli/mod.rs [hunk 2]
```diff
@@ -1,6 +1,263 @@
 //! `QuorumOS` client command line interface.
 //!
 //! See [`Command`] for all possible commands.
+//!
+//! ## Quorum Key Generation
```
Comment (jack-kearney) [inline qos-client/src/cli/mod.rs]: This whole process is going to be pretty complicated, I think this probably warrants having some kind of top-level descriptor that outlines all of the steps in the genesis process

File: qos-client/src/cli/mod.rs [hunk 3]
```diff
@@ -1,6 +1,263 @@
...
+//!
+//! - personal
+//!     - alice.our_namespace.setup.key
+//!     - alice.our_namespace.setup.pub
+//! - genesis
+//!     - alice.our_namespace.setup.pub
+//!     - bob.our_namespace.setup.pub
+//!     - eve.our_namespace.setup.pub
+//!
+//! Given the above directory structure, Alice can now generate the genesis
```
Comment (jack-kearney) [inline qos-client/src/cli/mod.rs:95]: This implies already having a machine with an enclave enabled, an enclave with the current version of QOS running inside of it, etc. Think we should probably include things like this as a "setup" step at the top of the guide

File: qos-client/src/cli/mod.rs [hunk 4]
```diff
@@ -1,6 +1,263 @@
...
+//!     - bob.our_namespace.setup.key
+//!     - bob.our_namespace.share
+//!     - bob.our_namespace.personal.pub
+//!     - bob.our_namespace.personal.key
+//! - genesis
+//!     - attestation_doc.genesis
+//!     - output.genesis
+//!
+//! Now each quorum member has their decrypted personal key and a quorum share
+//! encrypted to their personal key. (TODO: the personal key should be password
```
Comment (jack-kearney) [inline qos-client/src/cli/mod.rs]: Personal keys could also (probably should also) be stored on a smart card.

File: qos-client/src/cli/mod.rs (hunk 2 above)
Comment (jack-kearney) [inline qos-client/src/cli/mod.rs]: Think we should also call out the people who will be involved in this process. The "leader" and the "approvers" feels consistent with the terms you use in the document -- just would be helpful to describe the role of each at the top

File: qos-client/src/cli/mod.rs [hunk 5]
```diff
@@ -1,6 +1,263 @@
...
+//!
+//! ```shell
+//! cargo run --bin qos-cli generate-manifest \
+//!    --genesis-dir  ~/qos/our_namespace/genesis \
+//!    --boot-dir ~/qos/our_namespace/boot \
+//!    --nonce 0 \
+//!    --namespace our_namespace \
+//!    --pivot-hash 0xf0f0f0f0f0f0f0 \
+//!    --restart-policy always \
+//!    --pcr0 0xf0f0f0f0f0f0f0 \
```
Comment (jack-kearney) [inline qos-client/src/cli/mod.rs:204]: I think we should be clear about where these come from -- not necessarily the version, but that these should be included in any new release of QOS. And should be independently verified by each of the quorum members

File: qos-client/src/cli/mod.rs (hunk 2 above)
Comment (emostov) [reply qos-client/src/cli/mod.rs]: df97c2c0a84bc28908cfeb411e21c75b753a0f38

File: qos-client/src/cli/mod.rs (hunk 1 above)
Comment (emostov) [reply qos-client/src/cli/mod.rs]: Yeah, this is bad. 

In general I think there is a tricky line here of documenting the CLI vs documenting the quorum  os protocol. I think the hardcore way to do this is document in the protocol on the endpoints in qos-core and the qos-core protocol module and then just purely document the CLI assuming the reader understands the aforementioned .. but for now I think I am just gonna try and shove whatever context seems relevant into these docs

Comment (emostov) [reply qos-client/src/cli/mod.rs]: d39163824f330bbfe199e67305df7089339b77f0

File: qos-client/src/cli/mod.rs (hunk 2 above)
Comment (emostov) [reply qos-client/src/cli/mod.rs]: d39163824f330bbfe199e67305df7089339b77f0

File: qos-client/src/cli/mod.rs (hunk 3 above)
Comment (emostov) [reply qos-client/src/cli/mod.rs:95]: e164cb3eb80b62db26889e9d456a7571bee7f214

File: qos-client/src/cli/mod.rs (hunk 5 above)
Comment (emostov) [reply qos-client/src/cli/mod.rs:204]: 1172b204b369e4c3f5d383792577ab19bc2e3367

**Pull request merged**